crates/swc_bundler/src/bundler/mod.rs

```diff
@@ -44,6 +44,9 @@ pub struct Config {
 
     /// Type of emitted module
     pub module: ModuleType,
+
+    /// If it's false, try to find the real path.
+    pub preserve_symlinks: bool,
```

Comment:
I think this should be `resolve_symlinks`, and should default to false.

Reply:
`pnpm` is special case and most users will use default config.

Reply:
I agree with it will slow down the speed.
but I'm not sure if we can preserve symlinks by default. 
I think resolving symlinks is the default behavior of [node](https://nodejs.org/api/cli.html#--preserve-symlinks)/[webpack](https://webpack.js.org/configuration/resolve/#resolvesymlinks)/[esbuild](https://esbuild.github.io/api/#preserve-symlinks).

Reply:
We don't offer this yet, so even other tools are doing by default we have chance to reconsider default behavior.

Also even though this is not intended, if we make this as default this'll be breaking change _if anyone relies on current resolving_. I doubt there are many but technically.

I'd like to follow suggestion to make this not-enabled by default, then consider separately to make it as default.

Reply:
Does this really fix pnpm issue?

I investigated a bit, and I found that

- pnpm uses hard links (https://pnpm.io/faq)
- hard links are real files, and all hard links are treated as file, not link
- canonicalize uses `realpath` on unix (https://doc.rust-lang.org/std/fs/fn.canonicalize.html)

Reply:
pnpm use symlink and hard links. hard links it for pnpm's global cache, and the sym link is to reduce the depth of node_modules
https://pnpm.io/blog/2020/05/27/flat-node-modules-is-not-the-only-way
for example in #2945
pnpm link
```
node_modules/is-even -> node_modules/.pnpm/is-even@1.0.0/node_modules/is-even
node_modules/.pnpm/is-even@1.0.0/node_modules/is-odd -> node_modules/.pnpm/is-odd@0.1.2/node_modules/is-odd
node_modules/.pnpm/is-odd@0.1.2/node_modules/is-number -> node_modules/.pnpm/is-number@3.0.0/node_modules/is-number
node_modules/.pnpm/is-number@3.0.0/node_modules/kind-of -> node_modules/.pnpm/kind-of@3.2.2/node_modules/kind-of
```
```
  DEBUG Resolve is-even from Real(
    "/spack-pnpm-bug/index.ts",
) for Node

  DEBUG Resolve is-odd from Real(
    "/spack-pnpm-bug/node_modules/.pnpm/is-even@1.0.0/node_modules/is-even/index.js",
) for Node

  DEBUG Resolve is-number from Real(
    "/spack-pnpm-bug/node_modules/.pnpm/is-odd@0.1.2/node_modules/is-odd/index.js",
) for Node

  DEBUG Resolve kind-of from Real(
    "/spack-pnpm-bug/node_modules/.pnpm/is-number@3.0.0/node_modules/is-number/index.js",
) for Node

  DEBUG Resolve is-buffer from Real(
    "/spack-pnpm-bug/node_modules/.pnpm/kind-of@3.2.2/node_modules/kind-of/index.js",
) for Node

```

Reply:
If then, canonicalize is fine, but shouldn't it be called by resolver?